crates/agglayer-types/src/lib.rs

```diff
@@ -325,10 +328,9 @@ impl LocalNetworkStateData {
 
         let l1_info_root = certificate.l1_info_root()?;
 
-        certificate
-            .bridge_exits
-            .iter()
-            .for_each(|e| self.exit_tree.add_leaf(e.hash()));
+        for e in certificate.bridge_exits.iter() {
+            self.exit_tree.add_leaf(e.hash())?;
```

Comment:
No need to revert the changes made before the failure?

Reply:
yeah for the rollback in case of failure one should use `make_multi_batch_header` which applies the certificate on a temporary state